mmcv/cnn/rfsearch/search.py

```diff
@@ -0,0 +1,250 @@
+                if isinstance(module, ConvRFSearchOp):
+                    self.config['structure'][name] = module.op_layer.dilation
+            write_to_json(
+                self.config,
+                os.path.join(
+                    work_dir,
+                    'local_search_config_step%d.json' %
+                    self.config['search']['step'],
+                ),
+            )
+        elif (self.config['search']['step'] +
+              1) == self.config['search']['max_step']:
+            self.search_estimate_only(model)
+
+    def search(self, model: nn.Module):
+        """estimate and search for RFConvOp.
+
+        Args:
+            model (nn.Module): pytorch model
+        """
+        for _, module in model.named_modules():
+            if isinstance(module, ConvRFSearchOp):
+                module.estimate()
+                module.expand()
+
+    def search_estimate_only(self, model):
+        for module in model.modules():
+            if isinstance(module, ConvRFSearchOp):
+                module.estimate()
+
+    def wrap_model(self,
+                   model: nn.Module,
+                   config: Dict,
+                   search_op: str = 'Conv2d',
+                   init_rates: Optional[int] = None):
+        """wrap model to support searchable conv op.
+
+        Args:
+            model (nn.Module): pytorch model
+            config (Dict): search config file
+            search_op (str):
+                the module that uses RF search. Defaults to 'Conv2d'.
+            init_rates (int, optional):
+                Set to other initial dilation rates. Defaults to None.
+        """
+        op = 'torch.nn.' + search_op
+        for name, module in model.named_children():
+            if isinstance(module, eval(op)):
+                if 1 < module.kernel_size[0] and \
+                    0 != module.kernel_size[0] % 2 or \
+                    1 < module.kernel_size[1] and \
+                        0 != module.kernel_size[1] % 2:
+                    moduleWrap = eval(search_op + 'RFSearchOp')(
+                        module, init_rates, config['search'], self.verbose)
+                    moduleWrap = moduleWrap.cuda()
+                    if self.verbose:
+                        logger.info('Wrap model %s to %s.' %
+                                    (str(module), str(moduleWrap)))
+                    setattr(model, name, moduleWrap)
+            elif isinstance(module, ConvRFSearchOp):
+                pass
+            else:
+                if self.config['search']['skip_layer'] is not None:
+                    if any([
+                            each in name
+                            for each in self.config['search']['skip_layer']
+                    ]):
+                        continue
+                self.wrap_model(module, config, search_op, init_rates)
+
+    def set_model(self,
+                  model: nn.Module,
+                  config: Dict,
```

Comment:
Can we remove this parameter from this method because `config` is an attribute of the Hook which can be accessed by `self.config`?